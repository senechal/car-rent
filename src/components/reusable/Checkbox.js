import { useCallback } from 'react';
import PropTypes from 'prop-types';
const Checkbox = ({ value, children, checked, ...props }) => {
  return (
    <label {...props} className="rac-checkbox-label" data-checkboxvalue={value} data-testid="checkbox">
      <input type="checkbox" className="rac-checkbox-input" value={value} checked={checked} readOnly />
      {children}
    </label>
  );
};
Checkbox.propTypes = {
  value: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
  checked: PropTypes.bool,
};
Checkbox.defaultProps = {
  checked: undefined,
};

const Group = ({ onChange, ...props }) => {
  const handleBubbleClick = useCallback((event) => {
    const { checkboxvalue: value } = event.target.dataset;
    if (value) {
      onChange(value);
    }
  }, [onChange]);
  return <section {...props} onClick={handleBubbleClick} />;
};

Group.propTypes = {
  onChange: PropTypes.func,
};
Group.defaultProps = {
  onChange: Function.prototype,
};

Checkbox.Group = Group;

export default Checkbox;
