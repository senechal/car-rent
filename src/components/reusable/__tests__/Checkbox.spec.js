import { render, fireEvent } from '@testing-library/react';
import Checkbox from '../Checkbox';

const { Group } = Checkbox;

describe('Testing Checkbox Components', () => {
  describe('Testing Tag', () => {
    it('should render correctly', () => {
      const { container } = render(<Checkbox value="value">label</Checkbox>);
      expect(container).toMatchSnapshot();
    });
    it('should render correctly with checked true', () => {
      const { container } = render(<Checkbox value="value" checked>label</Checkbox>);
      expect(container).toMatchSnapshot();
    });
    it('should render correctly with checked false', () => {
      const { container } = render(<Checkbox value="value" checked={false}>label</Checkbox>);
      expect(container).toMatchSnapshot();
    });
  });
  describe('Testing Group', () => {
    it('should render correctlys', () => {
      const { container } = render(<Group><Checkbox value="value">content</Checkbox></Group>);
      expect(container).toMatchSnapshot();
    });
    it('should call onClose when clicked', () => {
      const callback = jest.fn();
      const { getByTestId } = render(<Group onChange={callback}><Checkbox value="value">content</Checkbox></Group>);
      const tag = getByTestId('checkbox');
      fireEvent.click(tag);
      expect(callback).toBeCalledWith('value');
    });
  });
});