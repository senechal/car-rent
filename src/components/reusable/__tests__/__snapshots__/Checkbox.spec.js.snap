// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Checkbox Components Testing Group should render correctlys 1`] = `
<div>
  <section>
    <label
      class="rac-checkbox-label"
      data-checkboxvalue="value"
      data-testid="checkbox"
    >
      <input
        class="rac-checkbox-input"
        readonly=""
        type="checkbox"
        value="value"
      />
      content
    </label>
  </section>
</div>
`;

exports[`Testing Checkbox Components Testing Tag should render correctly 1`] = `
<div>
  <label
    class="rac-checkbox-label"
    data-checkboxvalue="value"
    data-testid="checkbox"
  >
    <input
      class="rac-checkbox-input"
      readonly=""
      type="checkbox"
      value="value"
    />
    label
  </label>
</div>
`;

exports[`Testing Checkbox Components Testing Tag should render correctly with checked false 1`] = `
<div>
  <label
    class="rac-checkbox-label"
    data-checkboxvalue="value"
    data-testid="checkbox"
  >
    <input
      class="rac-checkbox-input"
      readonly=""
      type="checkbox"
      value="value"
    />
    label
  </label>
</div>
`;

exports[`Testing Checkbox Components Testing Tag should render correctly with checked true 1`] = `
<div>
  <label
    class="rac-checkbox-label"
    data-checkboxvalue="value"
    data-testid="checkbox"
  >
    <input
      checked=""
      class="rac-checkbox-input"
      readonly=""
      type="checkbox"
      value="value"
    />
    label
  </label>
</div>
`;
